tests/utils/test_data_type_utils.py

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: Apache-2.0
+# https://spdx.dev/learn/handling-license-info/
+
+# Third Party
+import pytest
+import torch
+
+# Local
+from tuning.utils import data_type_utils
+
+dtype_dict = {
+    "bool": torch.bool,
+    "double": torch.double,
+    "float32": torch.float32,
+    "int64": torch.int64,
+    "long": torch.long,
+}
+
+
+def test_str_to_torch_dtype():
+    for t in dtype_dict.keys():
+        assert data_type_utils.str_to_torch_dtype(t) == dtype_dict.get(t)
+
+
+def test_str_to_torch_dtype_exit():
+    with pytest.raises(SystemExit):
```

Comment:
this should test for vlaueError and not exit , if we pass a string with wrong dtype , it gives ValueError

Reply:
@Ssukriti The function should `raise ValueError("Unrecognized data type of a torch.Tensor")` instead of [exit(-1)](https://github.com/tedhtchang/fms-hf-tuning/blob/863cc53b8508bdf4f53aed98c788d4f8585c75f1/tuning/utils/data_type_utils.py#L25)

Reply:
Yikes I did not realize the code was just logging valueerror `logger.error(" ValueError: Unrecognized data type of a torch.Tensor")` , but exiting without throwing error. Thank you, I will have someone change it